scripts/prepdocslib/integratedvectorizerstrategy.py

```diff
@@ -0,0 +1,197 @@
+                    source_context="/document/pages/*",
+                    mappings=[
+                        InputFieldMappingEntry(name="content", source="/document/pages/*"),
+                        InputFieldMappingEntry(name="embedding", source="/document/pages/*/vector"),
+                        InputFieldMappingEntry(name="sourcepage", source="/document/metadata_storage_name"),
+                    ],
+                ),
+            ],
+            parameters=SearchIndexerIndexProjectionsParameters(
+                projection_mode=IndexProjectionMode.SKIP_INDEXING_PARENT_DOCUMENTS
+            ),
+        )
+
+        skillset = SearchIndexerSkillset(
+            name=skillset_name,
+            description="Skillset to chunk documents and generate embeddings",
+            skills=[split_skill, embedding_skill],
+            index_projections=index_projections,
+        )
+
+        return skillset
+
+    async def setup(self, search_info: SearchInfo):
+        search_manager = SearchManager(
+            search_info=search_info,
+            search_analyzer_name=self.search_analyzer_name,
+            use_acls=self.use_acls,
+            use_int_vectorization=True,
+            embeddings=self.embeddings,
+            search_images=False,
+        )
+
+        if self.embeddings is None:
+            raise ValueError("Expecting Azure Open AI instance")
+
+        await search_manager.create_index(
+            vectorizers=[
+                AzureOpenAIVectorizer(
+                    name=f"{search_info.index_name}-vectorizer",
+                    kind="azureOpenAI",
+                    azure_open_ai_parameters=AzureOpenAIParameters(
+                        resource_uri=f"https://{self.embeddings.open_ai_service}.openai.azure.com",
+                        deployment_id=self.embeddings.open_ai_deployment,
+                    ),
+                ),
+            ]
+        )
+
+        # create indexer client
+        ds_client = search_info.create_search_indexer_client()
+        ds_container = SearchIndexerDataContainer(name=self.blob_manager.container)
+        data_source_connection = SearchIndexerDataSourceConnection(
+            name=f"{search_info.index_name}-blob",
+            type="azureblob",
+            connection_string=self.blob_manager.get_managedidentity_connectionstring(),
+            container=ds_container,
+            data_deletion_detection_policy=NativeBlobSoftDeleteDeletionDetectionPolicy(),
+        )
+
+        await ds_client.create_or_update_data_source_connection(data_source_connection)
+        print("Search indexer data source connection updated.")
+
+        embedding_skillset = await self.create_embedding_skill(search_info.index_name)
+        await ds_client.create_or_update_skillset(embedding_skillset)
+
+    async def run(self, search_info: SearchInfo):
+        if self.document_action == DocumentAction.Add:
+            files = self.list_file_strategy.list()
```

Comment:
we should ignore the .md5s in this case since we want to upload all the files regardles if a previous local run already did this. Not sure if we need to edit the local strategy to support this as an option?

Reply:
Created a new work item